datafusion/proto/src/to_proto.rs

```diff
@@ -980,115 +959,30 @@ impl TryFrom<&ScalarValue> for protobuf::ScalarValue {
                 })
             }
-            scalar::ScalarValue::List(value, boxed_field) => match value {
-                Some(values) => {
-                    if values.is_empty() {
-                        protobuf::ScalarValue {
-                            value: Some(protobuf::scalar_value::Value::ListValue(
-                                protobuf::ScalarListValue {
-                                    field: Some(boxed_field.as_ref().try_into()?),
-                                    values: Vec::new(),
-                                },
-                            )),
-                        }
-                    } else {
-                        let scalar_type = match boxed_field.data_type() {
-                            DataType::List(field) => field.as_ref().data_type(),
-                            unsupported => {
-                                return Err(Error::General(format!("Proto serialization error: {:?} not supported to convert to DataType::List", unsupported)));
-                            }
-                        };
+            scalar::ScalarValue::List(values, boxed_field) => {
+                let is_null = values.is_none();
 
-                        let type_checked_values: Vec<protobuf::ScalarValue> = values
-                            .iter()
-                            .map(|scalar| match (scalar, scalar_type) {
-                                (
-                                    scalar::ScalarValue::List(_, list_type),
-                                    DataType::List(field),
-                                ) => {
-                                    if let DataType::List(list_field) =
-                                        list_type.data_type()
-                                    {
-                                        let scalar_datatype = field.data_type();
-                                        let list_datatype = list_field.data_type();
-                                        if std::mem::discriminant(list_datatype)
-                                            != std::mem::discriminant(scalar_datatype)
-                                        {
-                                            return Err(Error::inconsistent_list_typing(
-                                                list_datatype,
-                                                scalar_datatype,
-                                            ));
-                                        }
-                                        scalar.try_into()
-                                    } else {
-                                        Err(Error::inconsistent_list_designated(
-                                            scalar,
-                                            boxed_field.data_type(),
-                                        ))
-                                    }
-                                }
-                                (scalar::ScalarValue::Boolean(_), DataType::Boolean) => {
```

Comment:
it is not clear why there was other type specific code to  serializing lists - after this PR the existing code for serializing other types are used